engine/execution/messages.go

```diff
@@ -11,38 +11,122 @@ import (
 // TODO(patrick): rm unaccessed fields
 type ComputationResult struct {
 	*entity.ExecutableBlock
+	*ExecutionResults
+	*AttestationResults
+
+	*flow.ExecutionReceipt
+}
+
+func NewEmptyComputationResult(
+	block *entity.ExecutableBlock,
+) *ComputationResult {
+	return &ComputationResult{
+		ExecutableBlock:    block,
+		ExecutionResults:   NewEmptyExecutionResults(block),
+		AttestationResults: NewEmptyAttestationResults(block),
+	}
+}
+
+func (cr *ComputationResult) CollectionResult(colIndex int) *ColResSnapshot {
+	if colIndex < 0 && colIndex > cr.ExecutedColCounter {
+		return nil
+	}
+	return &ColResSnapshot{
+		blockHeader: cr.Block.Header,
+		collection: &flow.Collection{
+			Transactions: cr.CollectionAt(colIndex).Transactions,
+		},
+		collectionIndex:        colIndex,
+		isSystemCollection:     colIndex == len(cr.CompleteCollections),
+		updatedRegisters:       cr.StateSnapshots[colIndex].UpdatedRegisters(),
+		readRegisterIDs:        cr.StateSnapshots[colIndex].ReadRegisterIDs(),
+		emittedEvents:          cr.Events[colIndex],
+		transactionResults:     cr.transactionResultsByCollectionIndex(colIndex),
+		executionSnapshot:      cr.StateSnapshots[colIndex],
+		serviceEvents:          cr.ServiceEvents[colIndex],
+		convertedServiceEvents: cr.ConvertedServiceEvents[colIndex],
+	}
+}
+
+// ExecutionResults captures artifacts of execution of block collections
+type ExecutionResults struct {
```

Comment:
maybe rename to BlockExecutionResults

Reply:
renamed at [b8ccd7b](https://github.com/onflow/flow-go/pull/4078/commits/b8ccd7b4712e45aa01942322001a63e01c3b9342)